source/plugin/commands/commandbuilding.rst

```diff
@@ -0,0 +1,200 @@
+
+.. code-block:: java
+
+    import org.spongepowered.api.command.Command;
+
+    public Command.Parameterized buildCommand(){
+        return Command
+            .builder()
+            .executor(new HelloWorldCommand())
+            .permission("myplugin.command.helloWorld")
+            .shortDescription(Component.text("Hello World Command"))
+            .build();
+    }
+
+Overview of the Command.Builder methods
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. list-table:: 
+    :widths: 25 75
+    :header-rows: 1
+
+    * - method
+      - Description
+    * - executor
+      - Defines the command logic. *Setting the executor is required* if no child commands are set.
+    * - parameters
+      - Sets the parameters for use on the executor
+    * - flags
+      - Sets the flags for use on the executor
+    * - permission
+      - Sets the permission that will be checked before using this command
+    * - shortDescription
+      - A short, one-line description of this command's purpose that will be displayed by the help system
+    * - extendedDescription
+      - Sets an extended description to use in longer help listings. Will be appended to the short description
+    * - children
+      - Sets the child commands of this command with their aliases
+    * - executionRequirements
+      - Sets the execution requirements for the command.
+    * - terminal
+      - Sets if the command can execute despite the parameters stating that they are mandatory
+    * - build
+      - Builds the command. After that you need to register the command
+
+Player-Only commands
+~~~~~~~~~~~~~~~~~~~~
+
+Sometimes it is necessary that only players can execute a command. Example ``/suicide`` command. 
+This is best done though the ``executionRequirements`` on the command's builder where we check for
+a target player on the command cause. 
+
+.. code-block:: java
+
+    public Command.Parameterized buildCommand(){
+        return Command
+            .builder()
+            .executor(new HelloWorldCommand())
+            .permission("myplugin.command.helloWorld")
+            .shortDescription(Component.text("Hello World Command"))
+            .executionRequirements(context -> context.cause().context().get(EventContextKeys.PLAYER).isPresent())
+            .build();
+    }
+
+.. note::
+
+    We recommend you to add an optional ``[player]`` argument to make the command *console-friendly*. 
+    Example ``/suicide [player]``
+
+    The easiest solution for this is to append a ``playerOrTarget`` command parameter (see :doc:`argumentparsing`) 
+    instead of providing a execution requirement.
+
+.. tip::
+
+    Often times command are put in as player-only as they require the location in the world the command was
+    executed from. Best practise would be to check for the target location using :javadoc:`EventContextKeys#LOCATION`
+    instead of the player as this would allow command blocks to run the command without specifing a player.
```

Comment:
add a comma after "the player"
specifing -> specifying

Reply:
Updated